src/codegen_call_result.cpp

```
@@ -0,0 +1,52 @@
+#include "ry/codegen.hpp"
+#include "ry/diagnostic.hpp"
+
+// ===== Builtin Result Methods (and_then, map) =====
+
+llvm::Value *CodeGen::emitBuiltinResult(const CallExpr &e) {
+    if (e.args.size() != 2) return nullptr;
+    bool isAndThen = (e.callee == "and_then");
+    bool isMap     = (e.callee == "map");
+    if (!isAndThen && !isMap) return nullptr;
+
+    llvm::Value *resultVal = emitExpr(*e.args[0]);
```

Clang (14.0.6)

[warning] variable 'resultVal' is not initialized

(cppcoreguidelines-init-variables)